File: src/main/java/fr/ynerant/leveleditor/api/editor/RawSprite.java
package fr.ynerant.leveleditor.api.editor;

import fr.ynerant.leveleditor.api.editor.sprites.Sprite;

public class RawSprite {
    public static transient final RawSprite BLANK = new RawSprite();
    private String category = "blank";
    private int index = 0;

    public static RawSprite create(Sprite spr) {
        RawSprite raw = new RawSprite();
        raw.category = spr.getCategory().getName();
        raw.index = spr.getIndex();
        return raw;
    }

    public String getCategory() {
        return category;
    }

    public int getIndex() {
        return index;
    }
}

File: src/main/scala/fr/ynerant/leveleditor/api/editor/RawSprite.scala
package fr.ynerant.leveleditor.api.editor

import fr.ynerant.leveleditor.api.editor.sprites.Sprite


object RawSprite {
	val BLANK = new RawSprite("blank", 0)

	def create(spr: Sprite): RawSprite = {
		new RawSprite(spr.getCategory.getName, spr.getIndex)
	}
}

case class RawSprite(var category: String, protected var index: Int) {
	if (category == null) category = "blank"

	def getCategory: String = category

	def getIndex: Int = index
}
